rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java

```diff
@@ -178,6 +177,7 @@ public void start() {
 
             refreshWorkTimer = new Timer("Refresh work for mining");
             refreshWorkTimer.schedule(new RefreshBlock(), DELAY_BETWEEN_BUILD_BLOCKS_MS, DELAY_BETWEEN_BUILD_BLOCKS_MS);
+            started = true;
```

Comment:
I have to disagree with this change. Suppose something fails in the initialization of the components and you haven't set `started = true` at the begging. When you call `stop()` it won't got further than 
```
if (!started) {
            return;	            
       }
```
and, for instance, `blockListener` will never be removed from the ethereum object.

I think a better solution would be to just check for not null before calling `refreshWorkTimer.cancel();` on the `stop()` method.

Reply:
Followed this advice, the changes are in the ammend